### ISSUE TITLE
Implicit grant needed in invite-via-email sample?

### ISSUE THREAD

**Issue by gopal-amlekar** [azure-ad-b2c/samples#322]
Hi,

Thanks for a huge collection of B2C policies. I was trying to use the invite via email sample and have some question.

Is it mandatory to have implicit grant enabled for the main web app (used for the redeem process) when using the invite users via email sample (samples/policies/invite-via-email/)?

Without implicit grant, I get _error unauthorized_client AADB2C90057 The provided application is not configured to allow the OAuth Implicit flow_ when hitting the redeem policy endpoint /B2C_1A_INV_redeem/oauth2/v2.0/authorize?client_id=xxx

Thanks,

**Comment by JasSuri** [comment]
If `id_token` is in the `response_type` parameter of the auth request, you always need implicit flow for *id_token* enabled.

**Comment by gopal-amlekar** [comment]
Thank you @JasSuri that helps. I wanted the invitation based user experience without using implicit flow. I think I will use the other examples and add some own logic or backend code to handle invitation codes or something. Thanks again for your help, all the samples in this repo are really useful.

**Comment by JasSuri** [comment]
Implicit flow is going to be needed irrespective of the Custom Policy you use. 

The requirement is if the **application** is requesting `response_type=id_token`, implicit flow must be enabled for `id_tokens`.

**Comment by gopal-amlekar** [comment]
Here is what I am planning to do:

Generate a invitation code / token in backend in our system and send a custom link to the user. 
That link will take the user to our app which will then trigger a custom policy for signing up. 
The custom signup policy will validate the invitation code and user's email via REST API technical profile calling APIs into our system.
This validation can also include sending OTP using the inbuilt AD B2C verification control (or using again a custom email with SendGrid or similar)
User then may be asked to add some information like name etc. and then their account created within AD B2C.
Finally another REST API call into our system to update user's status and invalidate the invitation code. This can be done by our app at the redirect URI or an additional REST API technical profile in the policy.

So, effectively, the user signs up but the sign up is validated against our backend system.

I hope this should work, if you have any thoughts / suggestions, please let me know.

Thanks,

**Comment by JasSuri** [comment]
It will work. 
The B2C Custom Policy itself has no relevance to implicit flow requirement.

**Comment by gopal-amlekar** [comment]
Hi @davidspiess 

I used a combination of our existing APIs and a front-end app to implement this.  

I create the user only in my backend database via API and generate a unique invitation id for the user and sent them an invite email. The email contains an invitation link which points to my frontend app along with the invitation ID in the URL path. The frontend app verifies the invitation ID by querying backend API and then redirects the user to sign-up policy (I used a custom policy). After the user signs up, the frontend again calls my API to update the user's status and record that the user has created / activated their account.

Hope it helps?
